### PR TITLE
KTOR-370 Fix status code response accepting any numeric value (#1882)

### PR DESCRIPTION
**Pull request by vladf2n**
**Subsystem**
ktor-http

**Motivation**
#1882 

**Solution**
Create a conditional to check if the status-code is a valid response (between 100 and 599)

### CODE REVIEW

File: ktor-http/ktor-http-cio/common/src/io/ktor/http/cio/HttpParser.kt [hunk 1]
```diff
@@ -204,6 +210,7 @@ private fun parseStatusCode(text: CharSequence, range: MutableRange): Int {
     return status
 }
 
+private fun statusOutOfRange(code: Int) = code < HTTP_STATUS_CODE_MIN_RANGE || code > HTTP_STATUS_CODE_MAX_RANGE
```
Comment (cy6erGn0m) [inline ktor-http/ktor-http-cio/common/src/io/ktor/http/cio/HttpParser.kt:213]: Status codes higher than 599 (e.g. 6xx) aren't allowed; however, some services may use it despite that. So I would relax this limitation and apply the [HTTP syntax limitation](https://tools.ietf.org/html/rfc7230#section-3.1.2) that is defined like this:
>      status-code    = 3DIGIT

**Review by cy6erGn0m — CHANGES_REQUESTED** [review]
Thanks for your contribution. It's fine in general except for the small note regarding the upper status code limit.

File: ktor-http/ktor-http-cio/common/src/io/ktor/http/cio/HttpParser.kt (hunk 1 above)
Comment (vladf2n) [reply ktor-http/ktor-http-cio/common/src/io/ktor/http/cio/HttpParser.kt:213]: Thank you so much for your feedback! I'll fix it now.

Comment (vladf2n) [reply ktor-http/ktor-http-cio/common/src/io/ktor/http/cio/HttpParser.kt:212]: Just made a new commit to fix this limitation.

**Review by cy6erGn0m — APPROVED** [review]

**Pull request merged**